src/plugins/techStackGeneric.nim

```diff
@@ -123,20 +123,21 @@ proc scanFile(filePath: string, category: string, subcategory: string) =
     scanFileStream(stream)
 
 proc getProcNames(): HashSet[string] =
+  ## Returns every Name value in files at `/proc/[0-9]+/status`.
+  ## This is the filename of each executable, truncated to 15 characters.
   result = initHashSet[string]()
   for kind, path in walkDir("/proc/"):
-    for ch in path.splitPath().tail:
-      try:
-        if ch notin "0123456789":
+    if kind == pcDir and path.lastPathPart().allIt(it in {'0'..'9'}):
+      let data =
+        try:
+          readFile(path / "status")
+        except:
           continue
-        let p_path = path / "status"
-        var data = p_path.readFile()
-        for line in data.split("\n"):
-          if "Name:" in line:
-            var name = line.split("Name:")[1].strip()
-            result.incl(name)
-      except:
-         continue
+      for line in data.split("\n"):
```

Comment:
Why not splitLines stdlib function?

Reply:
You can also read the file one line at a time if you use the file stream from fd_cache so that we don't need to read the full file and then split by lines